examples/sst2/train.py

```diff
@@ -42,23 +50,35 @@ def sigmoid_cross_entropy_with_logits(*, labels: Array, logits: Array) -> Array:
   return relu_logits - logits * labels + jnp.log1p(jnp.exp(neg_abs_logits))
 
 
-def get_initial_params_and_state(key, model):
+def get_initial_params_and_state(rng, model):
   """Returns randomly initialized parameters and a fresh model state."""
   token_ids = jnp.ones((2, 3), jnp.int32)
   lengths = jnp.ones((2,), dtype=jnp.int32)
-  variables = model.init(key, token_ids, lengths)
-  state, params = variables.pop('params')
-  return params, state
+  variables = model.init(rng, token_ids, lengths, deterministic=True)
+  model_state, params = variables.pop('params')
+
+  # Remove intermediates for training. Otherwise our model state will fill up
+  # with intermediate outputs (exported using self.sow() commands). This will
+  # cause model_state to have a new shape on each step, triggering a new trace.
+  if 'intermediates' in model_state:
```

Comment:
Why would `"intermediates"` be in `model_state`, but only sometimes?

The documentation says it's not created during `init`:
https://flax.readthedocs.io/en/latest/howtos/extracting_intermediates.html#store-intermediate-values-in-a-new-variable-collection

Reply:
This was a recent change, thanks :) We can remove that part now. (And yeah they are really always there, unless the user removed the attention mechanism from the model).